### ISSUE TITLE
Segmentation fault, macOS Big Sur

### ISSUE THREAD

**Issue by cmey** [JuliaImGui/ImPlot.jl#5]
```
julia> versioninfo()
Julia Version 1.5.3
Commit 788b2c77c1 (2020-11-09 13:37 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
```

```
(imgui) pkg> st
Status `~/Code/julia/imgui-temp/imgui/Project.toml`
  [5d785b6c] CImGui v1.77.1
  [e8efc688] ImPlot v0.1.1 `https://github.com/wsphillips/ImPlot.jl#master`
```

```
julia> using ImPlot

julia> include(joinpath(pathof(ImPlot), "..", "..", "demo", "demo.jl"))
┌ Error: GLFW ERROR: code VERSION_UNAVAILABLE msg: NSGL: The targeted version of macOS does not support OpenGL 3.0 or 3.1 but may support 3.2 and above
└ @ Main ~/.julia/packages/ImPlot/1vXdO/demo/demo.jl:17

signal (11): Segmentation fault: 11
in expression starting at /Users/cmeyer/.julia/packages/ImPlot/1vXdO/demo/demo.jl:21
glfwSetWindowUserPointer at /Users/cmeyer/.julia/artifacts/81dbbcaad7406eabd969e7044076c56d9b2174ed/lib/libglfw.3.3.dylib (unknown line)
CreateWindow at /Users/cmeyer/.julia/packages/GLFW/CBo9c/src/glfw3.jl:502
CreateWindow at /Users/cmeyer/.julia/packages/GLFW/CBo9c/src/glfw3.jl:499 [inlined]
CreateWindow at /Users/cmeyer/.julia/packages/GLFW/CBo9c/src/glfw3.jl:499
unknown function (ip: 0x13367e130)
jl_apply at /Users/julia/buildbot/worker/package_macos64/build/src/./julia.h:1690 [inlined]
do_call at /Users/julia/buildbot/worker/package_macos64/build/src/interpreter.c:117
eval_body at /Users/julia/buildbot/worker/package_macos64/build/src/interpreter.c:0
jl_interpret_toplevel_thunk at /Users/julia/buildbot/worker/package_macos64/build/src/interpreter.c:660
jl_toplevel_eval_flex at /Users/julia/buildbot/worker/package_macos64/build/src/toplevel.c:840
jl_parse_eval_all at /Users/julia/buildbot/worker/package_macos64/build/src/ast.c:913
jl_load_rewrite at /Users/julia/buildbot/worker/package_macos64/build/src/toplevel.c:914 [inlined]
jl_load at /Users/julia/buildbot/worker/package_macos64/build/src/toplevel.c:919
include at ./client.jl:457
jl_apply at /Users/julia/buildbot/worker/package_macos64/build/src/./julia.h:1690 [inlined]
do_call at /Users/julia/buildbot/worker/package_macos64/build/src/interpreter.c:117
eval_body at /Users/julia/buildbot/worker/package_macos64/build/src/interpreter.c:0
jl_interpret_toplevel_thunk at /Users/julia/buildbot/worker/package_macos64/build/src/interpreter.c:660
jl_toplevel_eval_flex at /Users/julia/buildbot/worker/package_macos64/build/src/toplevel.c:840
jl_toplevel_eval_flex at /Users/julia/buildbot/worker/package_macos64/build/src/toplevel.c:790
jl_toplevel_eval at /Users/julia/buildbot/worker/package_macos64/build/src/toplevel.c:849 [inlined]
jl_toplevel_eval_in at /Users/julia/buildbot/worker/package_macos64/build/src/toplevel.c:883
eval at ./boot.jl:331
eval_user_input at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:134
repl_backend_loop at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:195
start_repl_backend at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:180
#run_repl#37 at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:292
run_repl at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:288
#807 at ./client.jl:399
jfptr_YY.807_70113.clone_1 at /Applications/Julia-1.5.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
jl_apply at /Users/julia/buildbot/worker/package_macos64/build/src/./julia.h:1690 [inlined]
do_apply at /Users/julia/buildbot/worker/package_macos64/build/src/builtins.c:655
jl_f__apply at /Users/julia/buildbot/worker/package_macos64/build/src/builtins.c:669 [inlined]
jl_f__apply_latest at /Users/julia/buildbot/worker/package_macos64/build/src/builtins.c:705
#invokelatest#1 at ./essentials.jl:710 [inlined]
invokelatest at ./essentials.jl:709 [inlined]
run_main_repl at ./client.jl:383
exec_options at ./client.jl:313
_start at ./client.jl:506
jfptr__start_57081.clone_1 at /Applications/Julia-1.5.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
true_main at /usr/local/bin/julia (unknown line)
main at /usr/local/bin/julia (unknown line)
Allocations: 1445585 (Pool: 1444979; Big: 606); GC: 1
[1]    83300 segmentation fault  julia --project=.
```

**Comment by cmey** [comment]
`CImGui.jl`'s demo runs fine.

**Comment by cmey** [comment]
![Screen Shot 2020-12-26 at 6 05 57 PM](https://user-images.githubusercontent.com/1876264/103156038-02f94a80-47a5-11eb-988a-180adb67d40d.png)

**Comment by cmey** [comment]
Since the error message mentions GLFW, I thought the issue was maybe in https://github.com/JuliaGL/GLFW.jl but the example code there runs fine.

**Comment by cmey** [comment]
Found this discussion that was helpful: https://stackoverflow.com/questions/41120025/isnt-it-possible-to-run-opengl-3-0-on-macbok-pro-2015-mid-with-glfw3

Replacing
```
GLFW.WindowHint(GLFW.CONTEXT_VERSION_MAJOR, 3)
GLFW.WindowHint(GLFW.CONTEXT_VERSION_MINOR, 0)
```
by
```
GLFW.WindowHint(GLFW.CONTEXT_VERSION_MAJOR, 3)
GLFW.WindowHint(GLFW.CONTEXT_VERSION_MINOR, 3)
GLFW.WindowHint(GLFW.OPENGL_FORWARD_COMPAT, GL_TRUE)
GLFW.WindowHint(GLFW.OPENGL_PROFILE, GLFW.OPENGL_CORE_PROFILE)
```

solves this issue.

**Comment by wsphillips** [comment]
Should be fixed by #8
